react/BarContextProvider/index.jsx

```diff
@@ -6,16 +6,29 @@ import { CozyProvider } from 'cozy-client'
 import { Provider } from 'react-redux'
 import { WebviewIntentProvider } from 'cozy-intent'
 
-const BarContextProvider = props => {
-  if (!props.children) return null
+const BarContextProvider = ({
+  children,
+  store,
+  client,
+  f,
+  t,
+  lang,
+  webviewService
+}) => {
+  if (!children) return null
+
   return (
-    <Provider store={props.store}>
-      <CozyProvider client={props.client}>
-        <I18nContext.Provider
-          value={{ f: props.f, t: props.t, lang: props.lang }}
-        >
+    <Provider store={store}>
+      <CozyProvider client={client}>
+        <I18nContext.Provider value={{ f, t, lang }}>
           <BreakpointsProvider>
-            <WebviewIntentProvider>{props.children}</WebviewIntentProvider>
+            {webviewService ? (
+              <WebviewIntentProvider webviewService={webviewService}>
```

Comment:
nit: I don't like having special condition for only one provider. Can't we just manage that inside the provider itself? OK we have a gain of not having a new wrapper, but does it come with a cost?

Reply:
Here I specifically do not want to create a WebviewIntentProvider because it itself creates its own context value if no props are given. We don't want that to ever happen because it would mean a cozy-bar could have a webview context while the parent app doesn't (or didn't pass the context as props)